### DISCUSSION TITLE
Can I conditionally exit the parent fzf process from an execute command?

### DISCUSSION THREAD

**Discussion by bschlenk** [junegunn/fzf#3760] [Q&A] [answered]
I've been using fzf to build a git branch script. One feature is that while viewing a list of branches in fzf, you can press the right arrow to jump into a commit view for that branch. This works by using `--bind right:execute(...)` and calling my script again with different args. By using `execute`, control returns to the branch view when the commit view exits.

Now I want to add additional things to the commit view, like binding ctrl-c to copy the sha. In this case I'd like to exit the script completely, bypassing the original fzf process. I can't figure out a way to do this from fzf directly. It seems to ignore exit codes from the `execute` process. I tried a hacky idea of passing the original process' pid to the child and then sending a SIGINT from the child, but that causes fzf not to clean things up and the terminal is busted.

Is there a way to do this without completely restructuring my script? I can switch to using `--expect` and starting the sub fzf window myself, but I like that `execute` uses a separate window and doesn't otherwise affect my terminal layout.

**Comment by junegunn** [comment, marked as answer]
You can start fzf with `--listen` and send `abort` action to the server listening on `$FZF_PORT` via POST request.

**Comment by bschlenk** [comment]
Thanks! I got something working using `--listen` but it feels pretty hacky. I'm sending `become(echo "{message}"; exit 7)` and then handling an exit code of 7 specially from the python script. The other issue is that anything I print from the second fzf window is lost when that window closes, so I need to send the message back up to the caller to print it out there instead.
